snap/squashfs/squashfs.go

```diff
@@ -87,14 +93,20 @@ func (s *Snap) Install(targetPath, mountDir string) (bool, error) {
 		return didNothing, nil
 	}
 
-	// try to (hard)link the file, but go on to trying to copy it
-	// if it fails for whatever reason
-	//
-	// link(2) returns EPERM on filesystems that don't support
-	// hard links (like vfat), so checking the error here doesn't
-	// make sense vs just trying to copy it.
-	if err := osLink(s.path, targetPath); err == nil {
-		return false, nil
+	overlayRoot, err := isRootWritableOverlay()
+	if err != nil {
+		logger.Noticef("cannot determine if root filesystem on overlay: %v", err)
+	}
+	if overlayRoot == "" {
```

Comment:
I think this deserves a comment:
```
// Hard-linking on overlayfs is identical to a full blown copy.
// When we are operating on a overlayfs based system (e.g. live installer)
// use symbolic links.
// https://bugs.launchpad.net/snapd/+bug/1867415
```
Feel free to either splice here or I can send one as a follow-up.